lib/starknet_explorer/data.ex

```diff
@@ -125,6 +125,55 @@ defmodule StarknetExplorer.Data do
     {:ok, tx}
   end
 
+  def full_transaction(tx_hash, network) do
+    tx =
+      case Transaction.get_by_hash_with_receipt(tx_hash) do
+        nil ->
+          {:ok, tx} = Rpc.get_transaction(tx_hash, network)
+          {:ok, receipt} = Rpc.get_transaction_receipt(tx_hash, network)
+
+          calldata = Calldata.from_plain_calldata(tx["calldata"])
+
+          block_id =
+            if receipt["block_number"],
+              do: %{"block_number" => receipt["block_number"]},
+              else: "latest"
+
+          input_data =
+            Enum.map(
+              calldata,
+              fn call ->
+                input =
+                  case Rpc.get_class_at(block_id, call.address, network) do
+                    {:ok, class} ->
+                      Enum.find(
+                        class["abi"],
```

Comment:
this fails with the Cairo0 ABIs, since those are encoded.
You could check if class["abi"] is map, if not then call to `Jason.decode/1`.